lib/streamlit/runtime/forward_msg_queue.py

```diff
@@ -166,3 +166,13 @@ def _maybe_compose_deltas(old_delta: Delta, new_delta: Delta) -> Delta | None:
         return new_delta
 
     return None
+
+
+def _update_script_finished_message(msg: ForwardMsg) -> ForwardMsg:
+    # When messages have been deleted from the queue
+    # from the perspective of the front end, the
+    # script which generated those messages were finished
+    # early for rerun.
```

Comment:
@dannyopts since I have just seen that you pushed again and before we trigger the merge gates, could you please push one last update to the docs here?

```python
# When we are here, the message queue is cleared from non-lifecycle messages
# before they were flushed to the browser. If there were no non-lifecycle messages in the queue,
# changing the type here should not matter for the frontend anyways, so we optimistically change the `script_finished`
# message to `FINISHED_EARLY_FOR_RERUN`. This indicates to the frontend that the previous run was interrupted by a new script start. Otherwise, a `FINISHED_SUCCESSFULLY` message might trigger a reset of widget states on the frontend.
```

Something along those lines are probably more descriptive, what do you think? After this change, we are ready to merge 🚀

Reply:
Ah and perhaps as a docstring comment

```python
"""
...
"""
```
since we are in a function body 🙂

Reply:
Done